src/bin/pgcopydb/schema.c

```diff
@@ -3377,122 +3226,90 @@ schema_list_partitions(PGSQL *pgsql,
-			"     from pgcopydb_table_size "
-			"     where oid = $2 "
-			"	union all "
-			"	select 1 as parts "
-			"	order by parts desc "
-			"	limit 1 "
-			" ), "
-			" ranges(n, parts, a, b) as "
-			" ( "
-			"   select n, "
-			"          parts, "
-			"          x as a, "
-			"          coalesce((lead(x, 1) over(order by n)) - 1, max) as b "
-			"     from key_bounds, t, "
-			"          generate_series(min, max, ((max-min+1)/parts)::bigint + 1) "
-			"          with ordinality as s(x, n) "
-			" ) "
-			" "
-			"  select n, parts, a, b, b-a+1 as count "
-			"    from ranges "
-			"order by n";
-
-		appendPQExpBuffer(sql, sqlTemplate,
-						  table->partKey, table->partKey, table->qname);
+		min = 0;
+		max = table->relpages;
 	}
 
-	if (PQExpBufferBroken(sql))
+	/*
+	 * Below code block calculates the number of parts needed and assigns the minimum and
+	 * maximum values for each part. It also logs information about each partition and
+	 * adds the table part to the catalog if provided.
+	 * Example:
+	 * int64_t tableSize (table->bytes) = 100;
+	 * int64_t partSize = 10;
+	 * int64_t min = 1;
+	 * int64_t max = 100;
+	 * int64_t result = partitionTable(&table, partSize, min, max, &catalog);
+	 * // Output:
+	 * // Partition table#1: 1 - 10 (10)
+	 * // Partition table#2: 11 - 20 (10)
+	 * // Partition table#3: 21 - 30 (10)
+	 * // ...
+	 * // Partition table#10: 91 - 100 (10)
+	 */
+	int64_t partsCount = ceil((double) table->bytes / (double) partSize);
+	int64_t range = ceil((double) (max - min + 1) / (double) table->bytes *
+						 (double) partSize);
+
+	for (int64_t i = 0; i < partsCount; i++)
 	{
-		(void) destroyPQExpBuffer(sql);
-		log_error("Failed to prepare partition query for table %s: out of memory",
-				  table->qname);
-		return false;
-	}
+		SourceTableParts *parts = &(table->partition);
 
-	int paramCount = 2;
-	Oid paramTypes[2] = { INT8OID, OIDOID };
-	const char *paramValues[2];
+		bzero(parts, sizeof(SourceTableParts));
 
-	paramValues[0] = intToString(partSize).strValue;
-	paramValues[1] = intToString(table->oid).strValue;
+		parts->partNumber = i + 1;
+		parts->partCount = partsCount;
+		parts->min = min + (i * range);
+		parts->max = min + ((i + 1) * range) - 1;
 
-	if (!pgsql_execute_with_params(pgsql, sql->data,
-								   paramCount, paramTypes, paramValues,
-								   &parseContext, &getPartitionList))
-	{
-		(void) destroyPQExpBuffer(sql);
-		log_error("Failed to compute partition list for table %s",
-				  table->qname);
-		return false;
-	}
+		if (parts->max > max)
+		{
+			parts->max = max;
+		}
 
-	(void) destroyPQExpBuffer(sql);
+		parts->count = parts->max - parts->min + 1;
 
-	if (!parseContext.parsedOk)
-	{
-		log_error("Failed to list table COPY partition list");
-		return false;
+		log_info("Partition %s#%d: %lld - %lld (%lld)", table->qname, parts->partNumber,
+				 (long long) parts->min, (long long) parts->max, (long
+																  long) parts->count);
```

Comment:
That should be `log_debug`. Maybe could be raised to `log_notice`, but I'm not even sure it's that important as a log message.